### PR TITLE
@OutParameter annotation for @SqlCall methods

### LINKED ISSUE #405: `@OutParameter` annotation
We should add a declarative annotation for registering out-parameters, as suggested by this user on the mailing list:

https://groups.google.com/d/msg/jdbi/yShp_-7Q7wM/8h99BCSqu5sJ

### PR DESCRIPTION
**Pull request by stevenschlansker**
Fixes #405

### CODE REVIEW

File: src/test/java/org/skife/jdbi/v2/sqlobject/TestOutParameters.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        OutParameters callStoredProc(@Bind("outParam1") Integer outParam1);
+
+        @SqlUpdate("CREATE OR REPLACE PROCEDURE procOneOUTParameter(outParam1 OUT NUMBER) IS BEGIN outParam1 := 100; END")
+        void insertStoredProc();
+    }
+
+    @SqlStatementCustomizingAnnotation(OutParameter.Factory.class)
+    @Target(ElementType.METHOD)
+    @Retention(RetentionPolicy.RUNTIME)
+    public @interface OutParameter {
```
Comment (qualidafial) [inline src/test/java/org/skife/jdbi/v2/sqlobject/TestOutParameters.java]: Could we add this add this annotation to the SQL Object API while we're at it?

**Comment by stevenschlansker** [comment]
@qualidafial After playing with this a bit and looking over the `Call` code, it looks like implementing this in v2 will not be easy.  The code is a bit of a mess and is clearly one of the least-loved corners of the codebase.  This is in no small part due to the frankly piss-poor design of the JDBC `CallableStatement` API.

I am thinking this might need to be deferred to v3, unless someone in particular needs v2 support and is willing to help implement it there.

**Comment by qualidafial** [comment]
Do you have a sense of how you would change the `Call` class in v3?

**Comment by stevenschlansker** [comment]
Well, at the very least, get rid of `SQLStatement` doing some `if (getClass().isSubclassOf(Call.class))` bullshit.  And I think the fact that OutParameters are registered as Parameters that don't ever actually bind to the statement is pretty bogus too..

**Comment by stevenschlansker** [comment]
PS @qualidafial I will be out for the next week in Michigan, so apologies if I'm not checking GitHub much.

**Comment by stevenschlansker** [comment]
I think I was wrong.  It's just not obvious how it works, but if you construct your SQL right (WTF is this `{call myStoredProc()}` BS?) it actually works fine.

**Comment by qualidafial** [comment]
Looks good.

**Pull request merged**